arwen/elrondapitest/bigFloats_test.go

```diff
@@ -0,0 +1,690 @@
+			value2 := big.NewFloat(0)
+			_ = value2.GobDecode(floatArgument2)
+			for i := 0; i < numberOfReps; i++ {
+				rDiv := big.NewInt(0)
+				value1.Int(rDiv)
+				result := big.NewFloat(0).Sub(value1, value2)
+				value1.Set(result)
+			}
+			floatBuffer, _ := value1.GobEncode()
+			verify.
+				Ok().
+				ReturnData(
+					floatBuffer)
+		})
+}
+
+func TestBigFloats_Abs(t *testing.T) {
+	bigFloatArguments := make([][]byte, numberOfReps+1)
+	for i := range bigFloatArguments {
+		bigFloatArguments[i] = make([]byte, 0)
+	}
+	bigFloatArguments[0] = repsArgument
+	for i := 0; i < numberOfReps; i++ {
+		floatValue := big.NewFloat(-float64(i) - 1)
+		encodedFloat, _ := floatValue.GobEncode()
+		bigFloatArguments[i+1] = encodedFloat
+	}
+
+	test.BuildInstanceCallTest(t).
+		WithContracts(
+			test.CreateInstanceContract(test.ParentAddress).
+				WithCode(test.GetTestSCCode("big-floats", "../../"))).
+		WithInput(test.CreateTestContractCallInputBuilder().
+			WithGasProvided(100000).
+			WithFunction("BigFloatAbsTest").
+			WithArguments(bigFloatArguments...).
+			Build()).
+		AndAssertResults(func(host arwen.VMHost, stubBlockchainHook *contextmock.BlockchainHookStub, verify *test.VMOutputVerifier) {
+			encodedLastFloat := bigFloatArguments[numberOfReps]
+			lastFloat := new(big.Float)
+			_ = lastFloat.GobDecode(encodedLastFloat)
+			absLastFloat := new(big.Float)
+			absLastFloat.Abs(lastFloat)
+			encodedAbsFloat, _ := absLastFloat.GobEncode()
+			verify.
+				Ok().
+				ReturnData(encodedAbsFloat)
+
+		})
+}
+
+func TestBigFloats_Neg(t *testing.T) {
+	bigFloatArguments := make([][]byte, numberOfReps+1)
+	for i := range bigFloatArguments {
+		bigFloatArguments[i] = make([]byte, 0)
+	}
+	bigFloatArguments[0] = repsArgument
+	for i := 0; i < numberOfReps; i++ {
+		floatValue := big.NewFloat(-float64(i) - 1)
+		encodedFloat, _ := floatValue.GobEncode()
+		bigFloatArguments[i+1] = encodedFloat
+	}
+
+	test.BuildInstanceCallTest(t).
+		WithContracts(
+			test.CreateInstanceContract(test.ParentAddress).
+				WithCode(test.GetTestSCCode("big-floats", "../../"))).
+		WithInput(test.CreateTestContractCallInputBuilder().
+			WithGasProvided(100000).
+			WithFunction("BigFloatNegTest").
+			WithArguments(bigFloatArguments...).
+			Build()).
+		AndAssertResults(func(host arwen.VMHost, stubBlockchainHook *contextmock.BlockchainHookStub, verify *test.VMOutputVerifier) {
+			encodedLastFloat := bigFloatArguments[numberOfReps]
+			lastFloat := new(big.Float)
+			_ = lastFloat.GobDecode(encodedLastFloat)
+			negLastFloat := new(big.Float)
+			negLastFloat.Neg(lastFloat)
+			encodedNegFloat, _ := negLastFloat.GobEncode()
+			verify.
+				Ok().
+				ReturnData(encodedNegFloat)
+		})
+}
+
+func TestBigFloats_Cmp(t *testing.T) {
+	bigFloatArguments := make([][]byte, 2*numberOfReps+1)
+	for i := range bigFloatArguments {
+		bigFloatArguments[i] = make([]byte, 0)
+	}
+	bigFloatArguments[0] = repsArgument
+	argsCounter := 1
+	for i := 0; i < numberOfReps; i++ {
```

Comment:
`numberOfReps = 1` because the assertions below only care about the last value returned by the contract.